### PR TITLE
Add metadata output tab for magic queries

### PR DESCRIPTION
**Pull request by michaelnchin**
Issue #, if available: #25

Description of changes:
- Display new output tab with query metadata when executing SPARQL and Gremlin magic queries.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/graph_notebook/magics/graph_magic.py [hunk 1]
```diff
@@ -163,37 +165,54 @@ def sparql(self, line='', cell='', local_ns: dict = None):
...
+            "# of results": "N/A",
+            "Response size (bytes)": "N/A",
+            "Results size (bytes)": "N/A"
+        }
 
         path = args.path if args.path != '' else self.graph_notebook_config.sparql.path
 
         logger.debug(f'using mode={mode}')
         if mode == QueryMode.EXPLAIN:
+            query_start = time.time() * 1000
```
Comment (austinkline) [inline src/graph_notebook/magics/graph_magic.py:183]: I'm guessing this makes `query_start` be measured in seconds? Can we add some documentation there? We probably want to report the latency of a query in milliseconds

File: src/graph_notebook/magics/graph_magic.py [hunk 2]
```diff
@@ -163,37 +165,54 @@ def sparql(self, line='', cell='', local_ns: dict = None):
         args = parser.parse_args(line.split())
         mode = str_to_query_mode(args.query_mode)
         tab = widgets.Tab()
+        titles = []
+        children = []
+
+        metadata = {
+            "Query mode": args.query_mode,
```
Comment (austinkline) [inline src/graph_notebook/magics/graph_magic.py:172]: Can we consider making a metadata class we instantiate, and have something that builds it into the output we want? It should help us minimize the amount of code we have to add to each magic command for us to have this type of data.

File: src/graph_notebook/magics/graph_magic.py [hunk 3]
```diff
@@ -232,8 +255,14 @@ def sparql(self, line='', cell='', local_ns: dict = None):
                 # of showing a tsv with each line being a result binding in addition to new ones.
                 if query_type == 'CONSTRUCT' or query_type == 'DESCRIBE':
                     lines = []
+                    bind_size = 0
```
Comment (austinkline) [inline src/graph_notebook/magics/graph_magic.py:258]: Are we using this still? Looks like its use was commented out later

**Review by austinkline — CHANGES_REQUESTED** [review]
Good start! I think we can do some structural changes to have a class that builds this metadata for us. Then when we want to add it to an output, we could call a method that returns the rendered html

File: src/graph_notebook/magics/graph_magic.py (hunk 1 above)
Comment (michaelnchin) [reply src/graph_notebook/magics/graph_magic.py:183]: Query_start and query_time are in milliseconds; time.time() returns current time in seconds as a high precision float so I multiplied by 1000 to convert to ms. I'll add some comments in the code with this info

**Pull request closed without merging**